### PR TITLE
Remove 1.x docs from develop branch

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
docs in `develop` branch

### Description
This PR removes all old 1.x docs in the `develop` branch to reduce confusion.
A hint where to find the 1.x docs is also added.

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: docs/content/setup/cloudron.md [hunk 1]
```diff
@@ -1,6 +0,0 @@
-Cloudron
```
Comment (DerMolly) [inline docs/content/setup/cloudron.md:1]: should this really be removed?

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -7,43 +7,15 @@ docs_dir: content
...
 nav:
   - Home: index.md
-  - Installation:
-      - 'Manual Installation': setup/manual-setup.md
-      - 'Reverse Proxy': setup/reverse-proxy.md
-      - Docker: setup/docker.md
-      - Cloudron: setup/cloudron.md
-      - Heroku: setup/heroku.md
-      - LinuxServer: setup/docker-linuxserver.md
-      - Yunohost: setup/yunohost.md
```
Comment (DerMolly) [inline docs/mkdocs.yml:17]: I would keep these

**Review by DerMolly — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

File: docs/mkdocs.yml (hunk 2 above)
Comment (davidmehren) [reply docs/mkdocs.yml:17]: Well, currently 2.0 does not support being installed in any useful way, so keeping any installation instructions is not very useful.

As soon as we have a way of combining frontend & backend etc., we can re-add installation instructions.

File: docs/content/setup/cloudron.md (hunk 1 above)
Comment (davidmehren) [reply docs/content/setup/cloudron.md:1]: see below

File: docs/mkdocs.yml (hunk 2 above)
Comment (DerMolly) [reply docs/mkdocs.yml:17]: okay

**Pull request merged**